### ISSUE TITLE
Package request: anyconnect

### ISSUE THREAD

**Issue by Pandapip1** [NixOS/nixpkgs#306647]
**Project description**

An enterprise VPN that my work means that I sometimes have to use.

**Metadata**

* homepage URL: https://www.cisco.com/site/us/en/products/security/secure-client/index.html
* source URL: https://software.cisco.com/download/home/286281283/type/282364313/release/4.10.08029
* license: proprietary
* platforms: unix, linux, darwin

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

**Comment by purepani** [comment]
You can use openconnect for this usually. It's what I've done anywhere that's had an anyconnect vpn.

**Comment by Pandapip1** [comment]
Openconnect doesn't support the way my organization has it configured. It keeps asking me for a username and password instead of sending me to the organization login page.

**Comment by purepani** [comment]
My organization also sends you to a login page, and I had issues getting it working until I changed the user agent to 'AnyConnect-compatible OpenConnect VPN agent' in the VPN settings, just in case that happens to work for you.

**Comment by Pandapip1** [comment]
The user-agent trick seems to have solved the issue. Thanks!

I will be leaving this package request open, however. It'd be nice to be able to have this in nixpkgs.

**Comment by Cu3PO42** [comment]
Unfortunately my organization is still using the older IPSec protocol with AnyConnect. This is, as far as I can tell, not supported by OpenConnect. The older vpnc does support this, but cannot authenticate via PKCS#11. At the current time, I do not think I can connect via anything but the proprietary client.

There is some prior work in #265443.

**Comment by cesarFuhr** [comment]
Hey, `openconnect` is not working with current employers configuration.
I would like this package too.

**Comment by TheToto** [comment]
I tried to make it work, but the app quit silently.

``` 
janv. 30 18:42:42 pc-nixos csc_ui[421279]: Function: setSignatureBuffer File: ../../vpn/Common/SignFile.cpp Line: 1829 Invoked Function: CCodeSignTlv::SetCodeSignTlv Return Code: -32440310 (0xFE11000A) Description: TLV_ERROR_INVALID_BUFFER
janv. 30 18:42:42 pc-nixos csc_ui[421279]: Function: Open File: ../../vpn/Common/SignFile.cpp Line: 203 Invoked Function: CSignFile::setSignatureBuffer Return Code: -32440310 (0xFE11000A) Description: TLV_ERROR_INVALID_BUFFER
janv. 30 18:42:42 pc-nixos csc_ui[421279]: Function: ExtractTLV_File File: ../../vpn/CommonCrypt/VerifyFileSignatureOpenSSL.cpp Line: 452 Invoked Function: CSignFile::Open Return Code: -32440310 (0xFE11000A) Description: TLV_ERROR_INVALID_BUFFER
janv. 30 18:42:42 pc-nixos csc_ui[421279]: Function: IsValid File: ../../vpn/CommonCrypt/VerifyFileSignatureOpenSSL.cpp Line: 154 Invoked Function: ExtractTLV_File Return Code: -32440310 (0xFE11000A) Description: TLV_ERROR_INVALID_BUFFER File (/nix/store/6ibc4qashr8mpv3syr0rhrn8pmhwlphi-cisco-secure-client/bin/vpnui)
```

**Comment by JeffDess** [comment]
I would also like that package. Company's VPN is Meraki AnyConnect with SAML, this is the only app on Linux I've managed to connect with (auth app must use a webview). I've spent some time trying to package it myself, but also hit signature errors similar to @TheToto on the systemd service (vpnd). I managed to build the client (vpnui), but it is crashing at launch if vpnd is not running. 

If this might help, there was also prior work made on the [Arch packager](https://github.com/ariel-miculas/cisco-secureclient]. Not Nix related, but it helped me outline the install steps I had to follow to write the derivation.

**Comment by Pandapip1** [comment]
This is specifically for Cisco AnyConnect

**Comment by JeffDess** [comment]
@Pandapip1 Yes, I was referring to Cisco Secure Client (which implements the AnyConnect protocol). This is the requested package right?

**Comment by Pandapip1** [comment]
Yes, I just wanted to make sure Meraki AnyConnect == Cisco AnyConnect

**Comment by JeffDess** [comment]
Exactly, Meraki is a Cisco network product, you can enable an AnyConnect VPN on it.

**Comment by JeffDess** [comment]
@Pandapip1 yes I guess I could open a PR for bumping unstable and patching stable? That would fix it for those unaware of the solution.

**Comment by senorsmile** [comment]
I had a possibly similar problem with openconnect.  For me, Network-manager would show the username/password for login fine, and then would show the 2 factor prompt.  However, there would be nowhere to enter the 2 factor code.  

I finally got it working by entering it from the cli: 
```nmcli connection show
# note the UUID of the connection you want

nmcli --ask connection up <UUID from above>
```

It then shows a lock icon in the network manager panel and check mark next to the connection under `VPN Connections`.
